app/grandchallenge/workstations/static/workstations/js/session-control.mjs

```diff
@@ -32,19 +34,17 @@ function openWorkstationSession(element) {
         }
 
         if (workstationWindow === null || isBlankContext) {
-            window.open(creationURI, windowIdentifier);
+            createNewSessionWindow(creationURI, windowIdentifier, element);
```

Comment:
I suggest replacing this with something along the lines: this keeps it more DRY:
```Javascript
function fallbackAction() {
   window.open(creationURI, windowIdentifier);
   removeSpinner(element);
}
...
fallbackAction();
...
fallbackAction();
```

Reply:
I'm already doing that here with the `createNewSessionWindow` function though, or what do you mean?

Reply:
You are, but you are repeating yourself in a way. If I would want to change the thing that happens when the session-control gives up advanced methods (i.e. post messages). I would need to change things in multiple locations.

Reply:
I'm confused. It already is just one location that you would need to update (the `createNewSessionWindow` function). What am I missing?

Reply:
It is somewhat of a style choice. So let's leave it. 

To go a bit deeper into my thoughts, just to clear the confusion: It works now, but imagine if you'd again need to add behavior to the fallback which would require additional arguments. You would then need to add that in two locations whereas, in the **inline**-function definition I suggested, you only need to do it at one spot.